From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Gianforcaro <b.gianfo@gmail.com>
Date: Mon, 27 Dec 2021 23:47:45 -0800
Subject: [PATCH] serenity: Fake O_SYNC for serenity so iomix stressor compiles

Serenity doesn't yet support O_SYNC, so just make it compile.
---
 stress-iomix.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/stress-iomix.c b/stress-iomix.c
index e99d262..8518e39 100644
--- a/stress-iomix.c
+++ b/stress-iomix.c
@@ -31,6 +31,11 @@
 #define MAX_IOMIX_BYTES		(MAX_FILE_LIMIT)
 #define DEFAULT_IOMIX_BYTES	(1 * GB)
 
+#if defined(__serenity__)
+    /* Serenity doesn't yet support O_SYNC, so just make it compile */
+    #define  O_SYNC (0)
+#endif
+
 typedef void (*stress_iomix_func)(const stress_args_t *args, const int fd, const off_t iomix_bytes);
 
 static const stress_help_t help[] = {
